### DISCUSSION TITLE
Komunikacja z serwerami Comarch/JPK z VM w Azure

### DISCUSSION THREAD

**Discussion by Sorendarchuk** [AzureCommunityPL/group#394] [Pomoc]
Cześć wszystkim,
mam problem z Comarchem po migracji VM do Azure (migracja była wykonana wgraniem VHD, później dograno aktualne bazy).
Całkowicie przestał działać moduł JPK ze względu na brak komunikacji z serwerami Comarch/JPK.
Zadaję pytanie na tym forum ponieważ wg. dokumentacji komunikacja musi być z serwerami znajdującymi wewnątrz infrastruktury Azure a dokładnie: 
produkcyjnym poprzez adres URL: https://e-dokumenty.mf.gov.pl/
należy odblokować go wraz z podstronami:
https://e-dokumenty.mf.gov.pl/api/Storage/InitUploadSigned 
https://e-dokumenty.mf.gov.pl/api/Storage/Status/{referenceNumber} 
Adresy kont storage-owych do których wysyłane są pliki JPK:
https://taxdocumentstorage01.blob.core.windows.net
https://taxdocumentstorage02.blob.core.windows.net
https://taxdocumentstorage03.blob.core.windows.net
https://taxdocumentstorage04.blob.core.windows.net
https://taxdocumentstorage05.blob.core.windows.net
https://taxdocumentstorage06.blob.core.windows.net
https://taxdocumentstorage07.blob.core.windows.net
https://taxdocumentstorage08.blob.core.windows.net
https://taxdocumentstorage09.blob.core.windows.net
https://taxdocumentstorage10.blob.core.windows.net
Z mojego rozpoznania wynika, że wszystkie wymienione adresy znajdują się w regionie West Europe (Netherlands). Moja maszyna znajduje się w Germany West Central więc to nie jest kwestia Service Endpointów w VNET.
Może ktoś podpowiedzieć czy w dobrym kierunku szukam, czy faktycznie problem może leżeć po stronie komunikacji wew. infrastruktury Azure? 
Przed migracją wszystko działało poprawnie, maszyna wcześniej była hostowana w private cloud.

**Comment by aszykm** [comment]
Nie napisałeś przez co łączy się do Internetu, czy jest wystawiony bezpośrednio na świat (czy są reguły NSG), czy łączy się przez NVA ?

**Reply by Sorendarchuk** [reply]
> Nie napisałeś przez co łączy się do Internetu, czy jest wystawiony bezpośrednio na świat (czy są reguły NSG), czy łączy się przez NVA ?

Łączy się przez NSG wykorzystując domyślne reguły ruchu wychodzącego.

**Comment by smereczynski** [comment]
Skąd się wziął ten Service Endpoint w Twojej wypowiedzi? Próbujesz użyć? Druga kwestia, pokaż NSG jeśli masz reguły wychodzące. Czy maszyna jest z Load Balancerem?

**Reply by Sorendarchuk** [reply]
> Skąd się wziął ten Service Endpoint w Twojej wypowiedzi? Próbujesz użyć? Druga kwestia, pokaż NSG jeśli masz reguły wychodzące. Czy maszyna jest z Load Balancerem?

Wziął się z tego, że kojarzę problem z komunikacją VM a Storage Accounts jeśli są w jednym regionie. Ale to nie mój case.
Reguła są defaultowe, brak load-balancera:
![image](https://user-images.githubusercontent.com/98914442/152517321-48ae191f-0425-457b-804d-2ef66800c7c8.png)

**Reply by smereczynski** [reply]
Traceroute do tych adresów robiłeś? Na samym sytemie operacyjnym nie ma jakiegoś firewalla który filtruje ten ruch?
